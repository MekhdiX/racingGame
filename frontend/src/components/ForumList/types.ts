import { ForumTopicResponse } from 'api/types';

export type Item = {
  id: number,
  name: string,
  answersCount: number,
  createdBy: string,
  date: string,
  category: string,
  active?: boolean,
};

export type Items = Item[];

export type OwnProps = {
  items: ForumTopicResponse
}
