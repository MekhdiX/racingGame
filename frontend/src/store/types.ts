import { Action } from 'redux';
import { ThunkAction, ThunkDispatch } from 'redux-thunk';
import store from 'store';

export type AppState = ReturnType<typeof store.getState>;
export type AppThunkDispatch = ThunkDispatch<AppState, unknown, Action>;
export type AppThunkAction<T> = ThunkAction<void, AppState, unknown, Action<T>>;
