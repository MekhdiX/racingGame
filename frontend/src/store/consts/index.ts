// Auth actions
export const SET_AUTH = 'SET_AUTH';
export const SIGN_IN = 'SIGN_IN';
export const SIGN_IN_YANDEX_OAUTH = 'SIGN_IN_YANDEX_OAUTH';
export const SIGN_UP = 'SIGN_UP';
export const LOG_OUT = 'LOG_OUT';

//User actions
export const SET_USER_INFO = 'SET_USER_INFO';
export const GET_USER = 'GET_USER';
export const CHANGE_USER_INFO = 'CHANGE_USER_INFO';
export const CHANGE_AVATAR = 'CHANGE_AVATAR';

// SSR actions
export const SET_SSR = 'SET_SSR';

// Theme actions
export const GET_THEME = 'GET_THEME';
export const SET_THEME = 'SET_THEME';

// Leaderboard actions
export const SET_LEADERBOARD = 'SET_LEADERBOARD';

// Feedback actions
export const SEND_FEEDBACK = 'SEND_FEEDBACK';

export const SET_FORUM_TOPIC = 'FETCH_FORUM_TOPIC';