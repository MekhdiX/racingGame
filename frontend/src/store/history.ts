import { createBrowserHistory, createMemoryHistory } from 'history';
import isServer from 'utils/isServerEnvCheker';

export default !isServer ? createBrowserHistory() : createMemoryHistory();
