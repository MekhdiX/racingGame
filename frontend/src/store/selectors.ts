import { AppState } from './types';

export const getAuthSelector = (state: AppState) => state.auth;

export const getUserSelector = (state: AppState) => state.user;

export const getSSRSelector = (state: AppState) => state.ssr;

export const getThemeSelector = (state: AppState) => state.theme;

export const getLeaderboardSelector = (state: AppState) => state.leaderboard;

export const getForumTopicsSelector = (state: AppState) => state.forumTopic;
