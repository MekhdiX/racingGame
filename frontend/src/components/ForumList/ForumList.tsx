import { ROUTES } from 'common/consts';
import { t } from 'common/dictionary';
import React, { FC } from 'react';
import { Link } from 'react-router-dom';
import { Table } from 'semantic-ui-react';

import { OwnProps } from './types';

const ForumList: FC<OwnProps> = ({ items }: OwnProps) => (
  <Table celled>
    <Table.Header>
      <Table.Row>
        <Table.HeaderCell>{t('question')}</Table.HeaderCell>
        <Table.HeaderCell />
        <Table.HeaderCell />
      </Table.Row>
    </Table.Header>

    <Table.Body>
      {items
        .map((item) => (
          <Table.Row key={item.id}>
            <Table.Cell>
              <Link to={ROUTES.FORUM_VIEW.replace(':id', item.id.toString())}>{item.title}</Link>
              <br/>
              <span>{item.description}</span>
            </Table.Cell>
            <Table.Cell>
              by <strong>{item.user}</strong>
            </Table.Cell>
          </Table.Row>
        ))}
    </Table.Body>
  </Table>
);
export default ForumList;
