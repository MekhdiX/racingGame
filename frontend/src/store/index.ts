import { createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import { routerMiddleware } from 'connected-react-router';
import history from './history';
import reducer from './reducers';
import { composeWithDevTools } from 'redux-devtools-extension';

const enhancer = applyMiddleware(thunk, routerMiddleware(history));

const store = createStore(reducer, composeWithDevTools(enhancer));

export default store;
