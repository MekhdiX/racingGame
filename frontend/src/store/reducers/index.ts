import { combineReducers } from 'redux';
import { connectRouter } from 'connected-react-router';

import history from '../history';
import auth from './auth';
import user from './user';
import ssr from './ssr';
import theme from './theme';
import leaderboard from './leaderboard';
import forumTopic from './forumTopic';

const reducer = combineReducers({
  router: connectRouter(history),
  auth,
  user,
  ssr,
  theme,
  leaderboard,
  forumTopic
});

export default reducer;
