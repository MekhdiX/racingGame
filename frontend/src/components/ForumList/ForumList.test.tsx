import ForumList from 'components/ForumList';
import { ConnectedRouter } from 'connected-react-router';
import * as React from 'react';
import { Provider } from 'react-redux';
import * as renderer from 'react-test-renderer';
import store from 'store';
import history from 'store/history';

describe(`ForumList Component`, () => {
  it(`renders correctly`, () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <ConnectedRouter history={history}>
            <ForumList
              items={[
                {
                  id: 1,
                  user: 'USER',
                  title: 'TITLE',
                  description: 'DESCR',
                  comments: [],
                },
              ]}
            />
          </ConnectedRouter>
        </Provider>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
