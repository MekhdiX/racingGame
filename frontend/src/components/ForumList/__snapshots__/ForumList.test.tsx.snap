// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForumList Component renders correctly 1`] = `
<table
  className="ui celled table"
>
  <thead
    className=""
  >
    <tr
      className=""
    >
      <th
        className=""
      >
        Вопрос
      </th>
      <th
        className=""
      />
      <th
        className=""
      />
    </tr>
  </thead>
  <tbody
    className=""
  >
    <tr
      className=""
    >
      <td
        className=""
      >
        <a
          href="/forum/1"
          onClick={[Function]}
        >
          TITLE
        </a>
        <br />
        <span>
          DESCR
        </span>
      </td>
      <td
        className=""
      >
        by 
        <strong>
          USER
        </strong>
      </td>
    </tr>
  </tbody>
</table>
`;
